crates/video/src/video_backends/v4l2_decoder.rs

```diff
@@ -0,0 +1,845 @@
+                    }
+
+                    v4l2r::bindings::v4l2_frmsizetypes_V4L2_FRMSIZE_TYPE_CONTINUOUS
+                    | v4l2r::bindings::v4l2_frmsizetypes_V4L2_FRMSIZE_TYPE_STEPWISE => {
+                        if frame.type_
+                            == v4l2r::bindings::v4l2_frmsizetypes_V4L2_FRMSIZE_TYPE_CONTINUOUS
+                            && format_frame.width.step != 1
+                            && format_frame.length.step != 1
+                        {
+                            warn!("invalid step for continious framesize");
+                            break;
+                        }
+
+                        format_frame.append_frame_rates(&mut self.video_enum_frame_intervals(
+                            pixformat,
+                            frame.__bindgen_anon_1.stepwise.max_width,
+                            frame.__bindgen_anon_1.stepwise.max_height,
+                        ));
+                    }
+
+                    _ => {
+                        warn!("Unexpected frame type: {}", frame.type_);
+                        break;
+                    }
+                }
+            }
+
+            frames.push(format_frame);
+            index += 1;
+        }
+
+        frames
+    }
+
+    fn video_enum_frame_intervals(
+        &self,
+        pixformat: u32,
+        width: u32,
+        height: u32,
+    ) -> Vec<video::virtio_video_format_range> {
+        let mut index: u32 = 0;
+        let mut frame_rates: Vec<video::virtio_video_format_range> = Vec::new();
+        loop {
+            let pixelformat = PixelFormat::from(pixformat);
+            let mut frate = video::virtio_video_format_range::default();
+            let ival: v4l2_frmivalenum = match v4l2r::ioctl::enum_frame_intervals(
+                &self.video_device,
+                index,
+                pixelformat,
+                width,
+                height,
+            ) {
+                Ok(frmivalenum) => frmivalenum,
+                Err(e) => {
+                    warn!("enum_frame_intervals failed! {}", e);
+                    break;
+                }
+            };
+
+            if index != ival.index {
+                warn!("driver returned wrong ival index: {}", ival.index);
+            }
+
+            if pixformat != ival.pixel_format {
+                warn!(
+                    "driver returned wrong ival pixel format: {:#x}",
+                    ival.pixel_format
+                );
+            }
+
+            if width != ival.width {
+                warn!("driver returned wrong ival width: {}", ival.width);
+            }
+
+            if height != ival.height {
+                warn!("driver returned wrong ival heigth: {}", ival.height);
+            }
+
+            unsafe {
+                match ival.type_ {
+                    v4l2r::bindings::v4l2_frmivaltypes_V4L2_FRMIVAL_TYPE_DISCRETE => {
+                        frate.min = ival.__bindgen_anon_1.discrete.denominator.into();
```

Comment:
I assume these are anonymous unions? Is there any way that these names could be a bit more sensible?

Reply:
This comes from the definition in `v4l2r`. I could make a wrapper or try another library.

Reply:
Can't we send a PR upstream to improve the situation?

Reply:
I will try to send a patch for this, and see how it goes.

Reply:
Btw, after looking into this more in depth, I realized this is generated using [bindgen](https://docs.rs/bindgen/latest/bindgen/), so the naming was given. While this was probably obvious to others, I was not too aware of the fact.

I will try to send a PR for `bindgen`, or for `v4l2r` (as I can argue that it could indeed benefit from some getters for the anonymous union/struct members), but these may not land. So ideally, I wouldn't let that be a blocker for this PR.

Reply:
I guess bindgen cannot know a name (since there is none on the C side). So probably it would need to be a manual mapping within v4l2r. I do not know how that would be best done. It is not a blocker I think, but if it is a simple fix upstream, then we should consider it.

Reply:
Right, I tried with a proposal in `v4l2r`. Let's see how it goes :) I'll keep this thread open in the meantime.

Reply:
Done :)